### DISCUSSION TITLE
Login error to OpenGrok

### DISCUSSION THREAD

**Discussion by ranjantomtom** [oracle/opengrok#4130] [General]
Hi,

Getting error while login to OpenGrok

Error url
http://opengrok.tomtomgroup.com/j_security_check;jsessionid=CCF6796D422D3954DB772234AF63AD61

Below context added to /opt/tomcat/conf/server.xml

<Realm className="org.apache.catalina.realm.JNDIRealm" debug="99"
        connectionURL="ldap://ldap-nl.tomtomgroup.com:389"
        referrals="follow"
        userBase="OU=Employees,O=Teleatlas,C=Global"
        userSearch="(sAMAccountName={0})"
        userSubtree="true"
        roleBase="CN=DevTools,CN=Roles,O=Teleatlas,C=Global"
        roleName="cn"
        roleSubtree="true"
        roleSearch="(member={0})"

And also updated to web.xml file with below 

<security-constraint>
  <web-resource-collection>
    <web-resource-name>images</web-resource-name>
    <url-pattern>/images/*</url-pattern>
  </web-resource-collection>
</security-constraint>

<security-constraint>
  <web-resource-collection>
    <web-resource-name>Protected Area</web-resource-name>
    <url-pattern>/*</url-pattern>
  </web-resource-collection>
  <auth-constraint>
    <role-name>TomTom Employees</role-name>
  </auth-constraint>
</security-constraint>

<login-config>
  <auth-method>FORM</auth-method>
  <realm-name>Please use your ttg.global username and password to access Opengrok.</realm-name>
  <form-login-config>
    <form-login-page>/login.jsp</form-login-page>
    <form-error-page>/login_error.html</form-error-page>
  </form-login-config>
</login-config>

**Comment by ranjantomtom** [comment]
Please suggest the possible reason for the error.

**Comment by tarzanek** [comment]
I think this is out of our scope, you should be able to find help in tomcat ldap integration docs , shouldn't you?

**Comment by ranjantomtom** [comment]
We can close this case as there were some network issue in our infrastructure due to that issue occurred.
